File: 26_gradle_plugin/app/src/main/java/com/hencoder/a26_gradle_plugin/MainActivity.java
package com.hencoder.a26_gradle_plugin;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.SystemClock;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        long startTime = SystemClock.uptimeMillis();
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        long timePassed = SystemClock.uptimeMillis() - startTime;
        if (timePassed > 500) {
            Log.d("方法执行时间", "onCreate - " + timePassed);
        }
    }
}

File: 18_motionlayout_example/src/main/java/com/example/youtube/MainActivity.kt
package com.example.youtube

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onClick(view: View) {
        val target = "com.example.youtube.${(view as Button).text}"
        startActivity(Intent(this, Class.forName(target)))
    }
}
